lib/metasploit/framework/login_scanner/http.rb

```diff
@@ -246,7 +246,12 @@ def send_request(opts)
           begin
             cli.connect
             req = cli.request_cgi(opts)
-            res = cli.send_recv(req)
+            # Authenticate by default
+            res = if opts['authenticate'].nil? || opts['authenticate']
+                    cli.send_recv(req)
+                  else
+                    cli._send_recv(req)
```

Comment:
Top one:
Sends request, and if it gets a header, it will authenticate

Bottom one:
If it asks to authenticate, it won't do anything, it'll just return the initial response